app/controllers/discourse_data_explorer/query_controller.rb

```diff
@@ -227,6 +229,22 @@ def run
 
     private
 
+    def rate_limit_query_runs!
+      return if !is_api? && !is_user_api?
```

Comment:
I am guessing these come from `DefaultCurrentUserProvider`? I wonder if we can do something like add a method called `is_api_call?` that covers both of these? Just a suggestion, it is fine as is as well, just could potentially be confusing for future engineers since you have to know this is in the current user provider

Reply:
I was thinking along the same lines. It'd need to be a change in core, but at the same time I think maybe this should accompany some bigger work around API stuff. I know TGX has been thinking about this a lot. 🙂

Reply:
Sure no problem this is fine as is :)